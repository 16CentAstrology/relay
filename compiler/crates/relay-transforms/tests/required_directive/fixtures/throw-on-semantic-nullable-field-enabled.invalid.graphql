# expected-to-throw
# relay:feature-flag disallow_required_action_throw_on_semantically_nullable_fields

query TestQuery {
  me {
    # This field does not have @semanticNonNull, so THROW should fail when feature flag is enabled
    name @required(action: THROW)
  }
}
