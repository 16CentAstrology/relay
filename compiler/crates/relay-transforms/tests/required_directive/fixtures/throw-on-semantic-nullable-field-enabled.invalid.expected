==================================== INPUT ====================================
# expected-to-throw
# relay:feature-flag disallow_required_action_throw_on_semantically_nullable_fields

query TestQuery {
  me {
    # This field does not have @semanticNonNull, so THROW should fail when feature flag is enabled
    name @required(action: THROW)
  }
}
==================================== ERROR ====================================
✖︎ @required with action THROW is not allowed on fields that are semantically nullable. Fields that can legitimately be null should not use the THROW action.

  throw-on-semantic-nullable-field-enabled.invalid.graphql:7:28
    6 │     # This field does not have @semanticNonNull, so THROW should fail when feature flag is enabled
    7 │     name @required(action: THROW)
      │                            ^^^^^
    8 │   }
